src/instance.rs

```diff
@@ -131,38 +146,154 @@ impl<T> Py<T> {
+    //     self.call(py, (), None)
+    // }
+
+    // /// Calls a method on the object.
+    // ///
+    // /// This is equivalent to the Python expression `self.name(*args, **kwargs)`.
+    // pub fn call_method(
+    //     &self,
+    //     py: Python,
+    //     name: &str,
+    //     args: impl IntoPy<Py<PyTuple>>,
+    //     kwargs: Option<&PyDict>,
+    // ) -> PyResult<PyObject> {
+    //     name.with_borrowed_ptr(py, |name| unsafe {
+    //         let args = args.into_py(py).into_ptr();
+    //         let kwargs = kwargs.into_ptr();
+    //         let ptr = ffi::PyObject_GetAttr(self.as_ptr(), name);
+    //         if ptr.is_null() {
+    //             return Err(PyErr::fetch(py));
+    //         }
+    //         let result = PyObject::from_owned_ptr_or_err(py, ffi::PyObject_Call(ptr, args, kwargs));
+    //         ffi::Py_DECREF(ptr);
+    //         ffi::Py_XDECREF(args);
+    //         ffi::Py_XDECREF(kwargs);
+    //         result
+    //     })
+    // }
+
+    // /// Calls a method on the object with only positional arguments.
+    // ///
+    // /// This is equivalent to the Python expression `self.name(*args)`.
+    // pub fn call_method1(
+    //     &self,
+    //     py: Python,
+    //     name: &str,
+    //     args: impl IntoPy<Py<PyTuple>>,
+    // ) -> PyResult<PyObject> {
+    //     self.call_method(py, name, args, None)
+    // }
+
+    // /// Calls a method on the object with no arguments.
+    // ///
+    // /// This is equivalent to the Python expression `self.name()`.
+    // pub fn call_method0(&self, py: Python, name: &str) -> PyResult<PyObject> {
+    //     self.call_method(py, name, (), None)
+    // }
 }
 
-/// Retrieves `&'py` types from `Py<T>` or `PyObject`.
+/// Retrieves `&'py` types from `Py<T>` or `Py<PyObject>`.
 ///
 /// # Examples
-/// `PyObject::as_ref` returns `&PyAny`.
-/// ```
-/// # use pyo3::prelude::*;
-/// use pyo3::ObjectProtocol;
-/// let obj: PyObject = {
-///     let gil = Python::acquire_gil();
-///     let py = gil.python();
-///     py.eval("[]", None, None).unwrap().to_object(py)
-/// };
-/// let gil = Python::acquire_gil();
-/// let py = gil.python();
-/// assert_eq!(obj.as_ref(py).len().unwrap(), 0);  // PyAny implements ObjectProtocol
-/// ```
-///
 /// `Py<T>::as_ref` returns `&PyDict`, `&PyList` or so for native types, and `&PyCell<T>`
 /// for `#[pyclass]`.
 /// ```
 /// # use pyo3::prelude::*;
 /// use pyo3::ObjectProtocol;
-/// let obj: PyObject = {
+/// let obj: Py<PyObject> = {
 ///     let gil = Python::acquire_gil();
 ///     let py = gil.python();
-///     py.eval("[]", None, None).unwrap().to_object(py)
+///     py.eval("[]", None, None).unwrap().to_object(py).into()
 /// };
 /// let gil = Python::acquire_gil();
 /// let py = gil.python();
-/// assert_eq!(obj.as_ref(py).len().unwrap(), 0);  // PyAny implements ObjectProtocol
+/// assert_eq!(obj.as_ref(py).len().unwrap(), 0);  // PyObject implements ObjectProtocol
 /// ```
 pub trait AsPyRef: Sized {
```

Comment:
Now that the old `PyObject`  doesn't exist, I'm wondering if `AsPyRef` trait should be moved to be just a method on `Py<T>`. I don't think there is value in it being a trait any more.